index.html

```diff
@@ -35,83 +46,956 @@
+                </div>
+                <div id="note3" resource="#note3">
+                    <sup>3</sup>Extensions may use the runtime object without declaring <code>"externally_connectable"</code> in the manifest. However, an extension may only receive messages sent from hosted web pages if this key is declared. It may also be used to block messages from other extensions, except for those in the <code>"ids"</code> list.<br />
+                </div>
+
+                <br />
+            </section>
+
+
+
+            <section>
+                <h4 id="webidl">Expressing Availability in WebIDL</h4>
+                <section>
+                    <h4>Attribute: <code>Exposed</code></h4>To express the aforementioned conditional situations in which certain APIs may or may not be available due to context, the following contexts are used:<br />
+                    <ul>
+                        <li>Window - The primary context for most browser extension APIs. This applies to background, event, popup, options and custom contexts</li>
+                        <li>ContentScript - Applies to content script context</li>
+                    </ul>
+                    <br /></section>
+                <section>
+                    <h4>Extended Attribute: <code>CheckAnyPermissions</code></h4>To express the aforementioned conditional situations in which certain APIs may or may not be available due to expressed manifest keys or permissions, the new <a href="http://heycam.github.io/webidl/">WebIDL</a> extended attribute <code>"<a href="https://developer.mozilla.org/en-US/docs/Mozilla/WebIDL_bindings#CheckAnyPermissions">CheckAnyPermissions</a>"</code> is used.<br />
+                    <br />
+                </section>
+
+                <section>
+                    <h4>Example (contextMenus)</h4>
+                    <font style="background:yellow;">DISCUSS: It has been suggested that we cut these examples.</font><br />
+                    First, let's look at an example of how the <code>browser.contextMenus</code> object would be expressed in WebIDL. In this case, the <code>contextMenus</code> object is only available to the browser host contexts such as background, vent, popup, and options pages. Also, the <code>contextMenus</code> object requires the extension author to specify <code>"permissions":["tabs"]</code> in the manifest.<br />
+                    <br />
+                    1) First, we declare the methods for the <code>browser.contextMenus</code> object.<br />
+<pre class="def idl">
+[NoInterfaceObject]
+interface BrowserContextMenus {
+    unsigned short create(ContextMenuCreateProperties data, Function callback);
+    object onClicked(Function callback);
```

Comment:
We're going to have to figure out how to handle event listeners. This describes a function. For the short term, we should probably go with something generic like `EventListener onClicked;`, and define the `EventListener` type.